dual_net.py

```diff
@@ -157,13 +103,93 @@ def __init__(self,
         self.update_op = tf.train.AdamOptimizer(learning_rate).minimize(self.loss)
```

Comment:
Regarding line 99 above:

```
self.policy_loss = tf.reduce_sum(tf.multiply(self.pi, self.policy_predict))
```

 don't we need a log somewhere?

![image](https://user-images.githubusercontent.com/10425351/33228645-c099e982-d174-11e7-9396-b714bd1a9568.png)

Reply:
The log is "included" in the net output.  Otherwise we'd be taking logs of negative numbers.

Reply:
Doesn't the self.policy_predict output only a valid probability distribution (sum to 1 and all positive values)? 

x = tf.exp(policy_out) * legality_mask_placeholder
# Needs reshape to broadcast properly 
policy_out = x / tf.reshape(tf.reduce_sum(x, axis=1), shape=((tf.shape(x)[0],) + (1,)))

Aside: It's an interesting question whether the model could learn without the log.  I think it might be ok, but may be harder to train.

Reply:
Oh right, we're not using log_softmax anymore, so yes we need to log somewhere